Clang 3.7.0 compile error

Hi guys,

as I tried to compile Cuberite with Clang 3.7.0 x86_64 I got the following build problem:
```
[ 27%] Built target event_core
[ 27%] Building CXX object src/OSSupport/CMakeFiles/OSSupport.dir/File.cpp.o
/.../Cuberite/src/OSSupport/File.cpp:337:18: error: loop variable 'item' of type 'const std::basic_string<char>' creates a copy from type 'const std::basic_string<char>' [-Werror,-Wrange-loop-analysis]
        for (const auto item: Contents)
                        ^
/.../Cuberite/src/OSSupport/File.cpp:337:7: note: use reference type 'const std::basic_string<char> &' to prevent copying
        for (const auto item: Contents)
             ^~~~~~~~~~~~~~~~
1 error generated.
src/OSSupport/CMakeFiles/OSSupport.dir/build.make:134: recipe for target 'src/OSSupport/CMakeFiles/OSSupport.dir/File.cpp.o' failed
CMakeFiles/Makefile2:1090: recipe for target 'src/OSSupport/CMakeFiles/OSSupport.dir/all' failed
Makefile:127: recipe for target 'all' failed
make[2]: *** [src/OSSupport/CMakeFiles/OSSupport.dir/File.cpp.o] Error 1
make[1]: *** [src/OSSupport/CMakeFiles/OSSupport.dir/all] Error 2
make: *** [all] Error 2
```

Based on the working Travis Build I suggest, that my clang has some more compile checks.
Btw.: I created my project with:
```
cmake -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_BUILD_TYPE=Debug ..
```

--- a/src/OSSupport/File.cpp
+++ b/src/OSSupport/File.cpp
@@ -334,7 +334,7 @@ bool cFile::DeleteFolder(const AString & a_FolderName)
 bool cFile::DeleteFolderContents(const AString & a_FolderName)
 {
 	auto Contents = cFile::GetFolderContents(a_FolderName);
-	for (const auto item: Contents)
+	for (const auto & item: Contents)
 	{
 		// Skip "." and ".." altogether:
 		if ((item == ".") || (item == ".."))

